torch_xla/core/dynamo_bridge.py

```diff
@@ -437,6 +486,8 @@ def optimized_mod(*args):
     result = res[len(xla_args_need_update):]
 
     none_remover.add_nones(result)
+    result = _maybe_move_tensors_to_device(result, original_device)
```

Comment:
ditto, only call this if `original_device ` is cuda

Reply:
ditto

Reply:
the problem of this `_maybe_move_tensors_to_device ` is it might force the execution to be blocking(since you need to move the result to cpu) instead of async. Prior to this change we just return in the bridge while execution happens in the background.

I want to isolate these potenial dangerous behavior(might cause unintended slowdown) so we don't accidentally introduce regressions. I also mentioned in my UX doc that we might need to find a way to convert the async xla tensor to a async cuda tensor to relief this issue.

Reply:
Updated to only do this if we have a cuda tensor.